### PR TITLE
Added Settings

### LINKED ISSUE #71: add opt-out for crashlytics
There should be an opt-out opportunity for users regarding crashlytics. Like a checkbox 'use anonymous data for crash analysis' on the start screen with a link to crashlytic's privacy policy below.

### PR DESCRIPTION
**Pull request by rhari991**
Work in progress
Will fix #71 , #68

### CODE REVIEW

**Comment by rhari991** [comment]
Only the update and crash reporting settings are pending. Will add them once pull requests #83  and #103 are merged

**Comment by Blaisorblade** [comment]
@rhari991 #83 and #103 are now merged ;-)

**Comment by jehy** [comment]
Oops, looks like you double reference some dependencies:

`

    compile 'com.github.clans:fab:1.6.4'
    provided "org.projectlombok:lombok:1.16.10"
    apt "org.projectlombok:lombok:1.16.10"

    // only include crashlytics in the online build
    onlineCompile('com.crashlytics.sdk.android:crashlytics:2.6.2@aar') {
        transitive = true;
    }

    compile 'com.github.clans:fab:1.6.4'
    provided "org.projectlombok:lombok:1.16.10"
    apt "org.projectlombok:lombok:1.16.10"
`

**Comment by jehy** [comment]
Also you double reference SettingsActivity in manifest:

`

        <activity
            android:name=".SettingsActivity"
            android:label="Settings"
            android:parentActivityName=".MainActivity"
            android:screenOrientation="portrait" />

        <activity android:name=".ShareHandlerActivity" android:theme="@android:style/Theme.Translucent.NoTitleBar">
            <intent-filter>
                <action android:name="android.intent.action.SEND" />
                <category android:name="android.intent.category.DEFAULT" />
                <data android:mimeType="image/*" />
            </intent-filter>
        </activity>

        <activity
            android:name=".SettingsActivity"
            android:label="Settings"
            android:parentActivityName=".MainActivity"
            android:screenOrientation="portrait" />
`

**Comment by rhari991** [comment]
uggh
I suck at merging

**Comment by rhari991** [comment]
Thanks for pointing that out !

**Comment by rhari991** [comment]
This should be good to go now.
One thing to note is that the version name has to be changed in three places in the build.gradle file when a new version is going to be released - in the defaultConfig and both the flavors.

**Comment by jehy** [comment]
Yup, such things happen when merging. Try using Android studio's integrated merge tool - it is much easier. Since you added it in your code, I am closing my pull request for checking flavor.

File: app/src/offline/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<manifest 
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <uses-permission 
+	android:name="android.permission.INTERNET"
+	tools:node="remove"/>
```
Comment (Blaisorblade) [inline app/src/offline/AndroidManifest.xml]: Got the change here (after looking up http://tools.android.com/tech-docs/new-build-system/user-guide/manifest-merger#TOC-tools:attr-markers), but this means the offline version currently isn't offline, right?

**Comment by Blaisorblade** [comment]
> One thing to note is that the version name has to be changed in three places in the build.gradle file when a new version is going to be released - in the defaultConfig and both the flavors.

Ugh... if that can't be avoided, can you add a comment in line so that it's harder to mess this up?
(But honestly, must the `versionName` be different in those cases?)

File: app/src/offline/AndroidManifest.xml (hunk 1 above)
Comment (rhari991) [inline app/src/offline/AndroidManifest.xml]: Yes.

**Comment by rhari991** [comment]
Good point, I was copying this from another project and didn't pay much attention to whether it was needed or not.

**Pull request merged**